docs/src/content/docs/api/webhooks.mdx

```diff
@@ -0,0 +1,492 @@
+---
+title: "Webhooks"
+sidebar:
+  order: 60
+tableOfContents: true
+---
+import { Badge, Tabs, TabItem } from '@astrojs/starlight/components';
+import CallOut from '@components/CallOut.astro';
+
+<CallOut
+  title="API Key Required"
+  text="The Ultravox API requires an API key. For more details see "
+  link={{ url: "../auth", text: "Authentication." }}
+/>
+<br />
+
+Webhooks allow you to receive real-time notifications about events in your Ultravox account. You can configure webhooks to send HTTP POST requests to a specified URL when certain events occur.
+
+## Webhooks in Ultravox
+When creating a webhook, you must provide a URL that will receive the webhook notification. Your service must return an HTTP status code of 200 ("OK"). If your service fails to respond with a 200, we will retry sending again.
+
+### Securing Webhooks
+You can optionally choose to secure your webhooks with a key. When creating a webhook, a secret key is automatically generated for you or you can choose to provide your own secret. You can update or patch your webhooks to change secrets in the event of a leak or as part of regular key rotation.
+
+Each time your server receives an incoming webhook from Ultravox here's how you can ensure the webhook was sent by Ultravox and hasn't been tampered with:
+
+1. **Timestamp Verification**
+* Each incoming webhook request includes a `X-Ultravox-Webhook-Timestamp` header with the time the webhook was sent.
+* Verify that this timestamp is recent (e.g. within the last minute) to prevent replay attacks.
+
+2. **Signature Verification**
```

Comment:
I think it'd be worthwhile to include the Python code sample I sent you to be as specific as possible.  When I read through [ngrok's study on common webhook implementations](https://ngrok.com/blog-post/get-webhooks-secure-it-depends-a-field-guide-to-webhook-security#:~:text=Use%20asymmetric%20keys%20for%20non,fetch%20new%20keys%20as%20needed.) one of the things they found most frustrating was docs with only English or pseudocode. Anything other than actual code leaves enough ambiguity that a bit-by-bit comparison of a hash signature is likely to fail in some way that by its nature yields no information about why.

Reply:
(Just as a simple example, this text doesn't mention that the date follows ISO formatting or that the signature is hex encoded)

Reply:
yeah good idea. adding.